File: algorithms/2_medium/I/00470_implement_rand10_using_rand7/solution.c
// The rand7() API is already defined for you.
// int rand7();
// @return a random integer in the range 1 to 7

int rand10() {
    while(true) {
        const int rows = rand7() - 1;
        const int cols = rand7() - 1;
        const int res = rows * 7 + cols;
        if (res < 40) {
            return (res / 4) + 1;
        }
    }
}

File: algorithms/2_medium/I/00470_implement_rand10_using_rand7/solution.rs
/** 
 * The rand7() API is already defined for you.
 * @return a random integer in the range 1 to 7
 * fn rand7() -> i32;
 */

impl Solution {
    pub fn rand10() -> i32 {
        loop {
            let rows = rand7() - 1;
            let cols = rand7() - 1;
            let res = rows * 7 + cols;
            if res < 40 {
                return (res % 10) + 1;
            }
        }
    }
}
